File: hu/codes/java/chapter_hashing/hash_map.java
/**
 * File: hash_map.java
 * Created Time: 2022-12-04
 * Author: krahets (krahets@163.com)
 */

package chapter_hashing;

import java.util.*;
import utils.*;

public class hash_map {
    public static void main(String[] args) {
        /* Hash tábla inicializálása */
        Map<Integer, String> map = new HashMap<>();

        /* Hozzáadási művelet */
        // Kulcs-érték pár (kulcs, érték) hozzáadása a hash táblához
        map.put(12836, "Xiao Ha");
        map.put(15937, "Xiao Luo");
        map.put(16750, "Xiao Suan");
        map.put(13276, "Xiao Fa");
        map.put(10583, "Xiao Ya");
        System.out.println("\nAfter adding is complete, hash table is\nKey -> Value");
        PrintUtil.printHashMap(map);

        /* Lekérdezési művelet */
        // Kulcs beírása a hash táblába az érték lekéréséhez
        String name = map.get(15937);
        System.out.println("\nInput student ID 15937, query name " + name);

        /* Törlési művelet */
        // Kulcs-érték pár (kulcs, érték) eltávolítása a hash táblából
        map.remove(10583);
        System.out.println("\nAfter removing 10583, hash table is\nKey -> Value");
        PrintUtil.printHashMap(map);

        /* Hash tábla bejárása */
        System.out.println("\nTraverse key-value pairs Key->Value");
        for (Map.Entry<Integer, String> kv : map.entrySet()) {
            System.out.println(kv.getKey() + " -> " + kv.getValue());
        }
        System.out.println("\nTraverse keys only Key");
        for (int key : map.keySet()) {
            System.out.println(key);
        }
        System.out.println("\nTraverse values only Value");
        for (String val : map.values()) {
            System.out.println(val);
        }
    }
}

File: hu/codes/python/chapter_hashing/hash_map.py
"""
File: hash_map.py
Created Time: 2022-12-14
Author: msk397 (machangxinq@gmail.com)
"""

import sys
from pathlib import Path

sys.path.append(str(Path(__file__).parent.parent))
from modules import print_dict

"""Fő kód"""
if __name__ == "__main__":
    # Hash tábla inicializálása
    hmap = dict[int, str]()

    # Hozzáadás
    # Kulcs-érték pár (kulcs, érték) hozzáadása a hash táblához
    hmap[12836] = "Xiao Ha"
    hmap[15937] = "Xiao Luo"
    hmap[16750] = "Xiao Suan"
    hmap[13276] = "Xiao Fa"
    hmap[10583] = "Xiao Ya"
    print("\nHozzáadás után a hash tábla:\nKulcs -> Érték")
    print_dict(hmap)

    # Lekérdezés
    # Kulcs beírása a hash táblába az érték lekérdezéséhez
    name: str = hmap[15937]
    print("\nA 15937-es diákigazolvány alapján a név: " + name)

    # Törlés
    # Kulcs-érték pár (kulcs, érték) törlése a hash táblából
    hmap.pop(10583)
    print("\nA 10583-as törlése után a hash tábla:\nKulcs -> Érték")
    print_dict(hmap)

    # Hash tábla bejárása
    print("\nKulcs-érték párok bejárása Kulcs->Érték")
    for key, value in hmap.items():
        print(key, "->", value)

    print("\nCsak kulcsok bejárása")
    for key in hmap.keys():
        print(key)

    print("\nCsak értékek bejárása")
    for val in hmap.values():
        print(val)
